cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -988,9 +988,9 @@ struct FailFunctor<VectorKernel::ChunkedExec> {
 };
 
 // GD for numeric types (integer and floating point)
-template <template <typename...> class Generator, typename Type0,
-          typename KernelType = ArrayKernelExec, typename... Args>
-KernelType GenerateNumeric(detail::GetTypeId get_id) {
+template <template <typename...> class Generator, typename Type0, typename... Args>
```

Comment:
Is this change necessary? I mean given that there are quite a few similar generators, it may worth to clean them up all together probably in an individual PR?

Reply:
Hmm, you're right that other generators also have an optional `KernelType` parameter. I could either refactor them all in this PR, or refactor them all in a separate PR (the former would be easier).

Reply:
OK, either way as you wish.